src/appleseed/renderer/modeling/bssrdf/directionaldipolebssrdf.cpp

```diff
@@ -99,12 +108,70 @@ namespace
             return Model;
         }
 
+        size_t compute_input_data_size(
+            const Assembly&         assembly) const APPLESEED_OVERRIDE
+        {
+            return align(sizeof(DirectionalDipoleBSSRDFInputValues), 16);
+        }
+
+        void prepare_inputs(void* data) const APPLESEED_OVERRIDE
+        {
+            DirectionalDipoleBSSRDFInputValues* values =
+                reinterpret_cast<DirectionalDipoleBSSRDFInputValues*>(data);
+
+            // Presets.
+            const size_t preset = floor(values->m_preset);
```

Comment:
That won't compile on Windows (and it certainly shouldn't on Linux!). Use

    const size_t preset = truncate<size_t>(values->m_preset);